spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -0,0 +1,273 @@
+    /// [`tzdb`]: https://docs.rs/tzdb/latest/tzdb/index.html
+    #[inline]
+    #[must_use]
+    pub fn tz(tz: TimeZoneRef<'static>) -> Self {
+        Self::Tz(tz)
+    }
+
+    /// Returns a `TimeZoneRef` which can be used to generate and project _time_.
+    #[inline]
+    #[must_use]
+    pub fn time_zone_ref(&'a self) -> TimeZoneRef<'a> {
+        match self {
+            Self::Utc => TimeZoneRef::utc(),
+            Self::Fixed(local_time_types) => match TimeZoneRef::new(&[], local_time_types, &[], &None) {
+                Ok(tz) => tz,
+                Err(_) => GMT,
+            },
+
+            Self::Tz(zone) => *zone,
+        }
+    }
+}
+
+impl From<&str> for Offset {
+    /// Construct a Offset based on the [accepted MRI values]
+    ///
+    /// Accepts:
+    ///
+    /// - `[+/-]HH[:]MM`
+    /// - A-I representing +01:00 to +09:00
+    /// - K-M representing +10:00 to +12:00
+    /// - N-Y representing -01:00 to -12:00
+    /// - Z representing 0 offset
+    ///
+    /// [accepted MRI values]: https://ruby-doc.org/core-2.6.3/Time.html#method-c-new
+    #[inline]
+    #[must_use]
+    fn from(input: &str) -> Self {
+        match input {
+            "A" => Self::fixed(1),
+            "B" => Self::fixed(2),
+            "C" => Self::fixed(3),
+            "D" => Self::fixed(4),
+            "E" => Self::fixed(5),
+            "F" => Self::fixed(6),
+            "G" => Self::fixed(7),
+            "H" => Self::fixed(8),
+            "I" => Self::fixed(9),
+            "K" => Self::fixed(10),
+            "L" => Self::fixed(11),
+            "M" => Self::fixed(12),
+            "N" => Self::fixed(-1),
+            "O" => Self::fixed(-2),
+            "P" => Self::fixed(-3),
+            "Q" => Self::fixed(-4),
+            "R" => Self::fixed(-5),
+            "S" => Self::fixed(-6),
+            "T" => Self::fixed(-7),
+            "U" => Self::fixed(-8),
+            "V" => Self::fixed(-9),
+            "W" => Self::fixed(-10),
+            "X" => Self::fixed(-11),
+            "Y" => Self::fixed(-12),
+            "Z" => Self::utc(),
+            _ => {
+                lazy_static! {
+                    static ref HH_MM_MATCHER: Regex = Regex::new(r"^([\-\+]{1})(\d{2})(\d{2})$").unwrap();
+                }
+                if HH_MM_MATCHER.is_match(input) {
+                    let caps = HH_MM_MATCHER.captures(input).unwrap();
+
+                    let sign = if caps.get(1).unwrap().as_str() == "+" { 1 } else { -1 };
+                    let hours = caps.get(2).unwrap().as_str().parse::<i32>().unwrap();
+                    let minutes = caps.get(3).unwrap().as_str().parse::<i32>().unwrap();
+
+                    let offset_seconds: i32 = sign * ((hours * SECONDS_IN_HOUR) + (minutes * SECONDS_IN_MINUTE));
+                    Self::fixed(offset_seconds)
+                } else {
+                    // TODO: ArgumentError
```

Comment:
that means this should probably be a `TryFrom` impl.

Reply:
this is a place where some REPL snippets would be handy inline comments:

```console
[3.1.2] > Time.new(2022, 3, 12, 15, 23, 59, "dead")
<internal:timev>:310:in `initialize': "+HH:MM", "-HH:MM", "UTC" or "A".."I","K".."Z" expected for utc_offset: dead (ArgumentError)
        from (irb):11:in `new'
        from (irb):11:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```

Reply:
Ah yes. I've got a local commit on my integration branch with this change. It's also passing a `&[u8]` instead since that's what's boxed. I'm leaving spinoso-time to convert that to a `&str` since Regex only likes `&str`s from my brief play

Reply:
sounds good to defer this to the errors PR